src/spotify.rs

```diff
@@ -1,227 +1,217 @@
-use aspotify::{
-	Album, Artist, Client, ClientCredentials, CountryCode, ItemType, Market, Playlist,
-	PlaylistItemType, Track, TrackSimplified,
-};
+use futures::{pin_mut, TryStreamExt};
 use librespot::core::authentication::Credentials;
 use librespot::core::cache::Cache;
 use librespot::core::config::SessionConfig;
 use librespot::core::session::Session;
+use rspotify::clients::BaseClient;
+use rspotify::model::{
+    AlbumId, ArtistId, Country, FullAlbum, FullArtist, FullPlaylist, FullTrack, IncludeExternal,
+    Market, PlayableItem, PlaylistId, SearchResult, SearchType, SimplifiedAlbum, SimplifiedTrack,
+    TrackId,
+};
+use rspotify::{ClientCredsSpotify, Credentials as ClientCredentials};
 use std::fmt;
 use std::path::Path;
 use url::Url;
 
 use crate::error::SpotifyError;
 
-
 pub struct Spotify {
-	// librespotify sessopm
-	pub session: Session,
-	pub spotify: Client,
-	pub market: Option<Market>,
+    // librespot session
+    pub session: Session,
+    pub spotify: ClientCredsSpotify,
+    pub market: Option<Market>,
 }
 
 impl Spotify {
-	/// Create new instance
-	pub async fn new(
-		username: &str,
-		password: &str,
-		client_id: &str,
-		client_secret: &str,
-		market_country_code: Option<CountryCode>,
-	) -> Result<Spotify, SpotifyError> {
-		// librespot
-		let cache = Cache::new(Some(Path::new("credentials_cache")), None, None, None).unwrap();
-		let credentials = match cache.credentials() {
-			Some(creds) => creds,
-			None => Credentials::with_password(username, password),
-		};
+    /// Create new instance
+    pub async fn new(
+        username: &str,
+        password: &str,
+        client_id: &str,
+        client_secret: &str,
+        market_country_code: Option<Country>,
```

Comment:
The country code does not parse correctly on line 56, and without a market many tracks fail to download

Reply:
`rspotify` parses ISO 3166-1 alpha-2 country codes instead of alpha-3 codes, and it seems to parse fine on my end.